testutils/channel.go

```diff
@@ -54,6 +54,11 @@ func NewServerChannel(opts *ChannelOpts) (*tchannel.Channel, error) {
 		return nil, fmt.Errorf("NewChannel failed: %v", err)
 	}
 
+	connOpts := ch.ConnectionOptions()
+	if opts.ConnContext != nil {
+		connOpts.ConnContext = opts.ConnContext
+	}
```

Comment:
not sure why this logic is here -- is it needed, and if so, can we put it in one of the existing functions